atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/SchemaMutationLock.java

```diff
@@ -147,89 +249,101 @@ private long waitForSchemaMutationLock() {
-                ByteBuffer rowName = ByteBuffer.wrap(globalDdlLockCell.getRowName());
-
-                List<Column> ourExpectedLock = ImmutableList.of(lockColumnWithValue(Longs.toByteArray(
-                        perOperationNodeIdentifier)));
-                Column clearedLock = lockColumnWithValue(Longs.toByteArray(
-                        CassandraConstants.GLOBAL_DDL_LOCK_CLEARED_VALUE));
-                CASResult casResult = writeLockWithCas(client, rowName, ourExpectedLock, clearedLock);
+            clientPool.runWithRetry((FunctionCheckedException<Cassandra.Client, Void, TException>) client -> {
+                int heartbeatCount = heartbeatService.heartbeatCount.get();
+                Column lockColumn = lockColumnWithStrValue(
+                        CassandraConstants.lockValueFromIdAndHeartbeat(perOperationNodeId, heartbeatCount));
+                List<Column> ourExpectedLock = ImmutableList.of(lockColumn);
+                Column clearedLock = lockColumnWithStrValue(CassandraConstants.GLOBAL_DDL_LOCK_CLEARED_VALUE);
+                CASResult casResult = writeDdlLockWithCas(client, ourExpectedLock, clearedLock);
 
                 if (!casResult.isSuccess()) {
-                    String remoteLock = "(unknown)";
-                    if (casResult.getCurrent_valuesSize() == 1) {
-                        Column column = Iterables.getOnlyElement(casResult.getCurrent_values(), null);
-                        if (column != null) {
-                            long remoteId = Longs.fromByteArray(column.getValue());
-                            remoteLock = (remoteId == CassandraConstants.GLOBAL_DDL_LOCK_CLEARED_VALUE)
-                                    ? "(Cleared Value)"
-                                    : Long.toString(remoteId);
-                        }
-                    }
-                    throw new IllegalStateException(String.format("Another process cleared our schema mutation lock"
-                            + " from underneath us. Our ID, which we expected, was %s, the value we saw in the"
-                            + " database was instead %s.", Long.toString(perOperationNodeIdentifier), remoteLock));
+                    handleForcedLockClear(casResult, perOperationNodeId, heartbeatCount);
                 }
-
                 LOGGER.info("Successfully released schema mutation lock.");
                 return null;
             });
-        } catch (Exception e) {
+        } catch (TException e) {
             throw Throwables.throwUncheckedException(e);
         }
     }
 
+    private void schemaMutationUnlockWithoutCas() {
+        schemaMutationLockForEarlierVersionsOfCassandra.unlock();
+    }
+
+    private void handleForcedLockClear(CASResult casResult, long perOperationNodeId, int heartbeatCount) {
+        Preconditions.checkState(casResult.getCurrent_valuesSize() == 1,
+                "Something is wrong with the underlying locks. Consult support for guidance.");
```

Comment:
Was going off of another exception message we already use in `waitForSchemaMutationLock`. That also uses "Consult support". But we can probably change both to "Contact support"?